protocols/v2/binary-sv2/no-serde-sv2/codec/src/datatypes/non_copy_data_types/mod.rs

```diff
@@ -1,3 +1,31 @@
+/// Provides a flexible, low-level interface for representing fixed-size and variable-size byte arrays,
+/// simplifying serialization and deserialization of cryptographic and protocol data.
+///
+/// The core component is the [`Inner`] type, a wrapper for managing both fixed and variable-length data slices
+/// or owned values. It offers aliases for commonly used data types like 32-byte hashes (`U256`), public keys 
+/// (`PubKey`), cryptographic signatures (`Signature`), and dynamically-sized arrays (`B0255`, `B064K`).
+
+/// # Features
+/// - **Fixed-size Aliases**: Types like [`U32AsRef`], [`U256`], [`ShortTxId`], [`PubKey`], and [`Signature`] 
+///   represent specific byte sizes, often used in cryptographic contexts or protocol identifiers.
+/// - **Variable-size Aliases**: Types like [`B032`], [`B0255`], [`Str0255`], [`B064K`], and [`B016M`] handle
+///   data with bounded sizes, providing flexibility for dynamic data.
+/// - **Traits and Conversions**: Implements traits like `From`, `TryFrom`, and [`IntoOwned`] for seamless
+///   transformations between owned and reference-based values.
+/// - **Property Testing** (optional, requires the `prop_test` feature): Supports generating arbitrary test data 
+///   for property-based testing.
+
+/// # Type Aliases
+/// - **[`U32AsRef`]**: 4-byte representation for small identifiers or integer values.
+/// - **[`U256`]**: 32-byte cryptographic hash (e.g., SHA-256 or protocol IDs).
+/// - **[`ShortTxId`]**: 6-byte transaction ID.
+/// - **[`PubKey`]**: 32-byte public key (e.g., Ed25519).
+/// - **[`Signature`]**: 64-byte cryptographic signature.
+/// - **[`B032`], [`B0255`], [`Str0255`]**: Variable-size representations for optional fields or protocol data.
+
+/// # Feature Flags
+/// - **`prop_test`**: Enables property-based testing with the `quickcheck` crate. When enabled, types like `U256`
+///   and `B016M` gain methods to generate arbitrary test data for testing serialization and deserialization.
```

Comment:
I cannot find a rendering of this header anywhere in the Rust Docs when I do `cargo doc`

no point in using `///` here